src/main/resources/com/google/api/codegen/ruby/version_index.snip

```diff
@@ -5,74 +5,142 @@
 @end
 
-@private tocModule(module, iterator)
+@private tocModule(indexType, module, iterator, content)
   @##
   @# # {@module.fullName} Contents
   @#
   @# | Class | Description |
   @# | ----- | ----------- |
-  @join content : module.contents
-    @# | [{@content.name}][] | {@content.description} |
+  @join class : module.contents
+    @# | [{@class.name}][] | {@class.description} |
   @end
   @#
-  @join content : module.contents
-    @# [{@content.name}]: {@content.link}
+  @join class : module.contents
+    @# [{@class.name}]: {@class.link}
   @end
   @#
-  {@simpleModule(module, iterator)}
+  {@simpleModule(indexType, module, iterator, content)}
 @end
 
-@private simpleModule(module, iterator)
+@private simpleModule(indexType, module, iterator, content)
   module {@module.moduleName}
     @if iterator.hasNext
-      {@modules(iterator)}
+      {@modules(iterator, indexType, content)}
+    @else
+      {@content}
     @end
   end
 @end
 
-@private generateReadme(metadata)
-  {@readme(metadata, exampleMethods(metadata.exampleMethods), installationLines(metadata))}
+@private generateReadme(indexType, metadata)
+  {@readme(metadata, exampleMethods(indexType, metadata.exampleMethods), installationLines(metadata))}
 @end
 
-@private exampleMethods(methods)
+@private exampleMethods(indexType, methods)
   @join method : methods on BREAK
     @#### {@method.apiClassName}
     ```rb
-    {@sampleCode_(method)}
+    @switch indexType.toString
+    @case "VersionIndex"
+      {@sampleCodeWithVersionImport(method)}
+    @case "TopLevelIndex"
+      {@sampleCodeWithTopLevelImport(method)}
+    @default
+    @end
     ```
   @end
 @end
 
 @private installationLines(metadata)
   $ gem install {@metadata.identifier}
 @end
+
+@private clients(index)
+  @join requireView : index.requireViews on BREAK.add(BREAK)
+    @if index.hasMultipleServices
+      module {@requireView.serviceName}
+        {@client(index, requireView)}
+      end
+    @else
+      {@client(index, requireView)}
+    @end
+  @end
+@end
+
+@private client(index, requireView)
+  @switch index.type.toString
+  @case "VersionIndex"
+    {@versionIndexClient(requireView)}
+  @case "TopLevelIndex"
+    {@topLevelClient(index, requireView)}
+  @default
+  @end
+@end
+
+@private versionIndexClient(requireView)
+  @##
+  @if @requireView.doc.lines
+    {@toComments(@requireView.doc.lines)}
+  @#
+  @end
+  {@initMethodComments()}
+  def self.new(*args, **kwargs)
+    {@requireView.clientName}.new(*args, **kwargs)
+  end
+@end
+
+@private topLevelClient(index, requireView)
+  @##
+  @if @requireView.doc.lines
+    {@toComments(@requireView.doc.lines)}
+  @#
+  @end
+  @# @@param version [Symbol, String]
+  @#   The major version of the service to be used. By default :{@index.apiVersion}
+  @#   is used.
+  {@initMethodComments()}
+  def self.new(*args, version: :{@index.apiVersion}, **kwargs)
+    case version.to_s.downcase
+    when "{@index.apiVersion}"
```

Comment:
Multiple versions will be supported by manually adding the versions to that case statement. 
This is because of a couple of limitations, the first being that toolkit only knows about a single version of the api and second that the tool manages the artifacts only manages the where the generated files go rather than making any edits to the files.